lib/resources/dstu2/medication_statement.yaml

```diff
@@ -0,0 +1,287 @@
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.asNeeded_x_
+
+  - name: route
+    required: 'No'
+    cardinality: 0..1
+    type: CodeableConcept
+    description: >
+                     A code specifying the route or physiological path of administration of a therapeutic
+                     agent into or onto a subject.
+    example: |
+              {"route": {
+                 "coding": [
+                   {"system": "http://snomed.info/sct",
+                    "code": "41974700",
+                    "display": "Chew - dosing instruction imperative (qualifier value)"}],
+                 "text": "Chewed"}}
+    binding:
+      description: >
+                      A coded concept describing the route or physiological path of administration of a
+                      therapeutic agent into or onto the body of a subject.
+      terminology:
+      - display: SNOMED CT
+        system: http://snomed.info/sct
+
+  - name: quantityQuantity
+    required: 'No'
+    cardinality: 0..1
+    type: SimpleQuantity
+    description: The amount of therapeutic or other substance given at one administration event.
+    example: |
+               {"quantityQuantity": {
+                  "value": 2,
+                  "unit": "tabs",
+                  "system": "http://unitsofmeasure.org",
+                  "code": "tbl"}}
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.quantity_x_
+
+- name: patient
+  required: 'Yes'
+  cardinality: 1..1
+  type: Reference (Patient)
+  description: Who is/was taking the medication.
+  example: '{"patient": {"reference": "Patient/5366327"}}'
+
+- name: status
+  required: 'Yes'
+  cardinality: 1..1
+  type: code
+  description: >
+                 A code representing the patient or other source's judgment about the state of the medication
+                 used that this statement is about.
+  example: '{"status": "active"}'
+
+- name: note
+  required: 'No'
+  cardinality: 0..1
+  type: string
+  description: Provides extra information about the medication statement that is not conveyed by the other attributes.
+  example: '{"note": "do not take with alcohol"}'
+  action: create
+
+- name: effectivePeriod
+  required: 'No'
+  cardinality: 0..1
+  type: Period
+  description: The interval of time during which it is being asserted that the patient was taking the medication.
+  note: If effectivePeriod and dosage.timing.repeat.boundsPeriod are both populated they need to be the same value.
+  example: |
+             {"effectivePeriod": {
+                "start": "2015-05-05T14:00:00.000Z",
+                "end": "2015-05-06T01:00:00.000Z"}}
+  url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.effective_x_
+  action: create
+
+- name: medicationReference
+  required: 'Yes'
+  cardinality: 1..1
+  type: contained Reference (Medication)
```

Comment:
For other resources where it can be a real Reference or a contained Reference we have labeled them as contained to make the distinction more obvious. While not technically incorrect it seems misleading, or at least non-obvious if we have to call it out somewhere else in the table, to list Reference (TypeA | TypeB | TypeC) where the types have different rules applied to them. I prompted Dayou to mark this as contained for consistency

https://github.com/cerner/fhir.cerner.com/blob/master/lib/resources/dstu2/allergy_intolerance.yaml#L89